test/relations.test.js

```diff
@@ -918,6 +918,79 @@ describe('relations', function() {
       };
     });
 
+    describe('find over related model with options', function() {
+      after(function() {
+        Physician.clearObservers('access');
+        Patient.clearObservers('access');
+      });
+      before(function() {
+        Physician.observe('access', beforeAccessFn);
+        Patient.observe('access', beforeAccessFn);
+
+        function beforeAccessFn(ctx, next) {
+          ctx.query.where.realm = ctx.options.realm;
+          next();
+        }
+      });
+      it('should find be filtered from option', function(done) {
+        var id;
+        Physician.create(function(err, physician) {
+          physician.patients.create({name: 'a', realm: 'test'}, function(err, ch) {
+            id = ch.id;
+            physician.patients.create({name: 'z', realm: 'test'}, function() {
+              physician.patients.create({name: 'c', realm: 'anotherRealm'}, function() {
+                verify(physician);
+              });
+            });
+          });
+        });
+
+        function verify(physician) {
+          physician.patients({}, {realm: 'test'}, function(err, records) {
+            if (err) return done(err);
+            should.exist(records);
+            records.length.should.eql(2);
+            records[0].realm.should.eql('test');
+            records[0].name.should.eql('a');
+            records[1].realm.should.eql('test');
+            records[1].name.should.eql('z');
+            done();
+          });
+        }
+      });
+
+      it('should find be filtered from option with promises', function(done) {
```

Comment:
IMO, there is no need to write this test twice (once for callback, second time for promises). The code fixing the issue is independent on the way how `find` was invoked. Let's keep only one of these tests.